src/test/java/org/opensearch/neuralsearch/query/HybridQueryInnerHitsIT.java

```diff
@@ -0,0 +1,467 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import lombok.SneakyThrows;
+import org.apache.lucene.search.join.ScoreMode;
+import org.junit.Before;
+import org.opensearch.index.query.InnerHitBuilder;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.index.query.NestedQueryBuilder;
+import org.opensearch.join.query.HasChildQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import org.opensearch.search.sort.SortOrder;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+import static org.opensearch.neuralsearch.util.TestUtils.DEFAULT_COMBINATION_METHOD;
+import static org.opensearch.neuralsearch.util.TestUtils.DEFAULT_NORMALIZATION_METHOD;
+
+public class HybridQueryInnerHitsIT extends BaseNeuralSearchIT {
```

Comment:
these tests are way heavier then integ tests should be, looks like you're trying different scenarios. Can you move this to unit tests, if not please minimize the number of model deployments. Inner hits are not specific to vector search queries like knn or neural, you can do multiple lexical queries where you don't need to deploy model.

Reply:
I haven't added any test regarding K-nn query for inner hits. neither I have deployed any model. Can you point me specific code piece?

Reply:
taking my previous comment back, I misread the code, there is no model deployment. I'm fine with existing test set